src/libs/SoloX.BlazorJsonLocalization/Services/Impl/AHttpHostedJsonLocalizationExtensionService.cs

```diff
@@ -71,9 +71,10 @@ protected AHttpHostedJsonLocalizationExtensionService(ILogger<AHttpHostedJsonLoc
             return await CultureInfoHelper.WalkThoughCultureInfoParentsAsync(cultureInfo,
                 cultureName =>
                 {
+                    var versionSuffix = !string.IsNullOrWhiteSpace(options.Version) ? $"?v={options.Version}" : string.Empty;
                     var uri = string.IsNullOrEmpty(cultureName)
-                    ? new Uri($"{basePath}.json", UriKind.Relative)
-                    : new Uri($"{basePath}{options.CultureSeparator}{cultureName}.json", UriKind.Relative);
+                    ? new Uri($"{basePath}.json{versionSuffix}", UriKind.Relative)
```

Comment:
That works just as well. I can change to that!

Reply:
Ok I see. In this case I would suggest something more customizable.
Let's say that we want to globally apply some file naming rules. The best way to do that could be to optionally provide a naming delegate.
In this case the default delegate would just be defined like this:

```csharp
(basePath, cultureName) => string.IsNullOrEmpty(cultureName)
      ? $"{basePath}.json"
       : $"{basePath}-{cultureName}.json";
```

This way the user would be able to implement its own naming policy with the assembly version or any global suffix.
It would also allows us to remove the `CultureSeparator` from the options.

Reply:
your implementation would be:

```csharp
options.NamingPolicy = (basePath, cultureName) => string.IsNullOrEmpty(cultureName)
     ? $"{basePath}.json?v={myVersion}"
     : $"{basePath}.{cultureName}.json?v={myVersion}";
```

Reply:
That's a great idea!

Reply:
or even more simple, the delegate could just be :

```csharp
file => $"{file}?v={myVersion}"
```

in this case we would keep the option `CultureSeparator`.

Note: its a bit less customizable but I think it's more readable....